designs/tls/install-cert-manager.md

```diff
@@ -0,0 +1,187 @@
+
+### User Stories
+
+#### Story 1
+
+As a user, I want to ensure that my data is securely transmitted between Mayastor components so that I can trust the integrity and confidentiality of my data during storage operations.
+
+#### Story 2
+
+As a user, I want to not have to worry about certificate creation or management.
+
+### Implementation Details/Notes/Constraints
+
+#### Cert Manager
+
+Cert-manager installation shall be done using [helm charts][certManagerHelmCharts]
+
+* Should there be discussion about drawbacks of depending on cert-manager or other strategies?
+
+#### Certificate Flow
+
+##### Example of certificate deployment with cert-manager
+
+1. With cert-manager installed we will use certificate and issuer resources to create root certificate
+
+```yaml
+{{- if .Values.base.tls.enabled }}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: selfsigned-issuer
+  namespace: {{ .Release.Namespace }}
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: root-ca
+  namespace: {{ .Release.Namespace }}
+spec:
+  isCA: true
+  duration: 175200h #20 years
+  commonName: root-ca
+  secretName: ca-root-cert
+  issuerRef:
+    name: selfsigned-issuer
+    kind: Issuer
+    group: cert-manager.io
+---
+# use this issuer to create other certs
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: ca-issuer
+  namespace: {{ .Release.Namespace }}
+spec:
+  ca:
+    secretName: ca-root-cert
+```
+
+In this example:
+
+* The `root-ca` certificate is created with `isCA: true`, making it a root certificate.
+
+* The `selfsigned-issuer` is used to issue the root certificate.
+
+* An `Issuer` named `ca-issuer` is created, which uses the `ca-root-cert` secret containing the root certificate.
+
+1. Use above to sign below, See issuerRef field.
+
+```yaml
+{{- if .Values.base.tls.enabled }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: rest-api-server
+  namespace: {{ .Release.Namespace }}
+spec:
+  duration: 175200h
+  isCA: false
+  dnsNames:
+    - {{ .Release.Name }}-api-rest.{{ .Release.Namespace }}.svc
+    - {{ .Release.Name }}-api-rest.{{ .Release.Namespace }}.svc.cluster.local
+    - {{ .Release.Name }}-api-rest
+  issuerRef:
+    name: ca-issuer
+    kind: Issuer
+  secretName: rest-api-server-cert
+{{- end }}
+```
+
+In this example:
+
+* The `rest-api-server` certificate is created and signed by the `ca-issuer`, which uses the root certificate.
+
+* The `isCA` field is set to `false`, indicating that this is not a CA certificate.
+
+> **_NOTE:_** We will utilize the secrets created to volumeMount certs on appropriate deployment or daemonset.
+
+#### HTTP as default
+
+Components should be implemented in such a way that they will communicate using HTTP if tls is not enabled and HTTPS if tls is enabled.
+
+#### etcd
+
+Will need direction on how to implement tls between etcd and it's communication partners
```

Comment:
Yes this would also require enabling TLS on the etcd chart
I think the etcd chart already supports this

Reply:
https://github.com/Johnaius/mayastor-extensions/blob/47b5a178d03491e67f61a7abc1711e7749fc7b9a/chart/values.yaml#L512-L517

Reply:
Another place for `TLS` is the agent/agent `gRPC` and agent/io-engine `gRPC`